src/string_search.h

```diff
@@ -44,7 +44,9 @@ class Vector {
 
   // Access individual vector elements - checks bounds in debug mode.
   T& operator[](size_t index) const {
+#ifdef DEBUG
     CHECK(index < length_);
```

Comment:
I'd keep the check - if it's worth doing in debug mode, it's almost always worth doing in release mode too.

I probably should have updated the comment in 1b7372f2fb5.  Can you do that now?